### ISSUE TITLE
Powerstate ack without change

### ISSUE THREAD

**Issue by maffmeier** [iobroker-community-adapters/ioBroker.sony-bravia#157]
**Describe the bug**  
Der Power Datenpunkt gibt Ack ohne Änderung. Dadurch werden z.B. Szenen ungewollt ausgelöst.

**To Reproduce**  
1. Adapter mit Fernseher verbinden
2. Datenpunkt als Szenen trigger verwenden. Status der Szene manuell überschreiben.
3. Szene wird nach Ack erneut gestartet.

**Expected behavior**  
Wenn sich der Powerstatus nicht unterscheidet zum vorherigen Update sollte auf das setzen des Datenpunktes verzichtet werden.

**Versions:**  
 - Adapter version: v1.0.9

**Comment by cooper2k4** [comment]
Meinst Du `info/powerStatusActive`? Ich beobachte da auch, dass sich der Wert während des Fernsehens häufig verändert ohne dass das Gerät ausgeschaltet wird und damit Aktionen getriggert werden.

**Comment by mcm1957** [comment]
Welches Problem liegt hier nun vor?

@cooper schreibt, dass sich der WERT ändert
@maffmeier schreibt dass der Satet "Ack gibt" - was immer demit gemei nt ist.

Ändert sich nun der Value (.val) des states unerwartet oder nicht ?

Das ein State aktualisisiert wird und das ggF auch mit demselben Wert stellt per se kein Problem dar. Falls der Bedarf besteht nur auf Werteänderungen zu triggern kann das ja z.B. in js so umgesetzt werden. ACK sollte beim Schreiben des States durch den Adapter jedenfalls immer gesetzt sein.

**Comment by maffmeier** [comment]
@cooper2k4 Ja ganu, den meine ich. 

Ich habe iobroker so verstanden, dass es ack nur geben soll, wenn sich der wert ändert. es kann sein, dass wie du es beschreibst auch bei mir der fall ist, dann passiert der werte wechsel zwischen true und false aber so schnell, dass ich es nicht mit bekomme. 
Ich werde mal das History auf dem Value aktivieren, dann werde ich es feststellen können.

**Comment by mcm1957** [comment]
Nein, der Adapter kann jederzeit identische Werte neu schreiben. Typisch machen dies z.B. Adapter die regelmäßig daten von externen Quellen holen. Ein Adapter der z.B. den Pegelstand eines Flusses ermittelt kann durchaus alls x Minuten denselben Wert schreiben. Dies ist völlig OK und zulässig.

ACK = true sagt nur, dass der Wert bestätigt ist. Das ist bei einem Wert den der Adapter selbst als Ausgabe schreibt (fast) immer der Fall. ACK = FALSE wird bei der Übergabe von Werten an einen Adapter z.B. via VIS oder Script verwendet um anzuzeigen, dass der Wert nicht bestätigt ist sondern an das Gerät erst übermittelt werden soll.

**Comment by maffmeier** [comment]
Okay, verstanden,
aber dieses verhalten sorgt dafür, dass eine szene immer wieder getriggert wird. Ich habe mir jetzt geholfen, in dem ich ein script mit einer "puffer"-variable gebaut habe.

**Comment by mcm1957** [comment]
> @cooper2k4 Ja ganu, den meine ich.
> 
> Ich habe iobroker so verstanden, dass es ack nur geben soll, wenn sich der wert ändert. es kann sein, dass wie du es beschreibst auch bei mir der fall ist, dann passiert der werte wechsel zwischen true und false aber so schnell, dass ich es nicht mit bekomme. Ich werde mal das History auf dem Value aktivieren, dann werde ich es feststellen können. EDIT: hier innerhalb kurzer zeit: <img alt="image" width="1118" src="https://private-user-images.githubusercontent.com/2385541/306749046-d3c128d9-784e-4ec3-83df-1bc6c6a34370.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1NDM4MDksIm5iZiI6MTcwODU0MzUwOSwicGF0aCI6Ii8yMzg1NTQxLzMwNjc0OTA0Ni1kM2MxMjhkOS03ODRlLTRlYzMtODNkZi0xYmM2YzZhMzQzNzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjFUMTkyNTA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjkwYTA4OWJlYTMwZGExYTM4MjkxYTg2YWVhOWU2ZWRhOWYwYzk4NDI3M2M1YzUzMDRjM2I1ZGY3NzEwYjkzNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.utFZg6a8eKbdk1bKydaDchS8T79uQqDTF_IeOhtZ9Og">

Offenbar leist der Adpater alle 10s den Wert und schreibt ihn in den State. Das ist per se zulässig. Es wäre aber ebenso zulässig, wenn der Adapter den Wert nur bei Änderungen schreibt. Vorteil ist weniger Schreiblast, Nachteil ist, dass man nicht weiß ob der Wert wirklich noch gültig ist oder schon lönger nicht gelesen werden konnte.

Das Problem bzw. der Wunsch ist mal klar.
HAB ABER IM Momengt keine AHnung wann sich das wer ansehen und ggF anpassen kann.
Danke

**Comment by mcm1957** [comment]
An sich sollte es gehen, dass das Script nur bei Werteänderung getriggert wird.

**Comment by maffmeier** [comment]
Gerne, ich konnte mir ja erstmal helfen.

Ein Vorschlag meiner Seits: Dass der Adapter vorher prüft ob sich der Wert im Objekt unterscheidet, so kann weiterhin alle 10 Sekunden geprüft werden, aber eben wie du schreibst weniger Schreiblast.

In meinem Konkretenfall, hatte ich es damals so gemacht, eben ohne skript:
<img width="879" alt="image" src="https://github.com/iobroker-community-adapters/ioBroker.sony-bravia/assets/2385541/d8399e34-92de-42cf-ae81-468ccd3bc755">

**Comment by sreevo** [comment]
Here the same 😐 Please correct this error

**Comment by mcm1957** [comment]
> Here the same 😐 Please correct this error

This behavior is NO error. It's completly valid to write identical values. See comments above.
Feel free to react on changed value and not an value written if required.
